sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -477,11 +477,19 @@ def run_inference(
     return predictions
 
   def get_num_bytes(self, batch: Sequence[Tuple[KeyT, ExampleT]]) -> int:
+    keys, unkeyed_batch = zip(*batch)
+    batch_bytes = len(pickle.dumps(keys))
     if self._single_model:
-      keys, unkeyed_batch = zip(*batch)
-      return len(
-          pickle.dumps(keys)) + self._unkeyed.get_num_bytes(unkeyed_batch)
-    return len(pickle.dumps(batch))
+      return batch_bytes + self._unkeyed.get_num_bytes(unkeyed_batch)
+
+    batch_by_key = defaultdict(list)
+    for pair in batch:
```

Comment:
nit: consider using name readability if pair can be unpacked: 

```
    for key, examples in batch:
      batch_by_key[key].append(examples)
```

Reply:
also there is conflation of batch of keyed prediction inputs vs a batch of elements for a single key.  not sure how to clarify, could use a courtesy variable like:

```    
    if self._single_model:
      return batch_bytes + self._unkeyed.get_num_bytes(unkeyed_batch)
    else:
      keyed_batches = batch

    for key, examples in keyed_batches:
      ...
```

Reply:
I updated to use key/examples.

> also there is conflation of batch of keyed prediction inputs vs a batch of elements for a single key. not sure how to clarify, could use a courtesy variable like:

I don't think this is right. `batches` means batch of keyed prediction inputs in all contexts here, `batch_by_key` represents batches of elements per key.